doc/api/assert.md

```diff
@@ -1063,6 +1063,31 @@ If the values are not strictly equal, an `AssertionError` is thrown with a
 `message` parameter is an instance of an [`Error`][] then it will be thrown
 instead of the `AssertionError`.
 
+## assert.stringMatching(input, match)
+<!-- YAML
+added: v10.11.1
+-->
+* `input` {string|string[]}
+* `match` {RegExp|RegExp[]|string|string[]}
+
+Verify that requested constraints match the provided input :
```

Comment:
This description doesn't explain how the function works or what it verifies.

Reply:
Maybe it was a bit too short ^^ The purpuse of the function is to see if input match the rules we set in `match` parameters. For example : check that a template string correspond to the format we want (for example "BUILD V{Number}" , valid : "BUILD V5" , invalid : "BUILD VA")

PS: I used to write parameters in RAML (a superset of YAML) that allows to tell that parameter can be an string or an array that contains several type of data (RegExp or string)